src/Disciplina/DB/DSL/Interpret/Sqlite3.hs

```diff
@@ -0,0 +1,79 @@
+
+module Disciplina.DB.DSL.Interpret.Sqlite3 where
+
+import Universum
+
+import Control.Monad.Reader
+import Control.Monad.Catch
+import Control.Monad.Trans
+
+import qualified Data.Set as Set (Set, singleton, member)
+import Database.SQLite.Simple
+
+import Disciplina.DB.DSL.Class
+
+newtype Sqlite3T m a = Sqlite3T { getSqlite3T :: ReaderT Connection m a }
+    deriving (MonadTrans)
+
+runSqlite3T :: (MonadIO m, MonadCatch m) => Sqlite3T m a -> String -> m a
+runSqlite3T connectionString action = do
+    connection <- liftIO $ open connectionString
+    res <- getSqlite3T action
+        `runReaderT` connection
+        `finally`    close connection
+    return res
```

Comment:
Also, yes, you don't need to open and close the database anywhere but in `*Resource` modules (where you do that using `ComponentM`).